src/kernel/lsm/ss_table/loader.rs

```diff
@@ -0,0 +1,278 @@
+        self.factory.exists(gen)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::hash_map::RandomState;
+    use std::sync::Arc;
+    use bincode::Options;
+    use bytes::Bytes;
+    use tempfile::TempDir;
+    use crate::kernel::io::{FileExtension, IoFactory, IoType};
+    use crate::kernel::lsm::log::LogLoader;
+    use crate::kernel::lsm::mem_table::{data_to_bytes, DEFAULT_WAL_PATH};
+    use crate::kernel::lsm::ss_table::loader::SSTableLoader;
+    use crate::kernel::lsm::storage::Config;
+    use crate::kernel::lsm::version::DEFAULT_SS_TABLE_PATH;
+    use crate::kernel::Result;
+    use crate::kernel::utils::lru_cache::ShardingLruCache;
+
+    #[test]
+    fn test_ss_table_loader() -> Result<()> {
+        let temp_dir = TempDir::new().expect("unable to create temporary working directory");
+
+        let value = Bytes::copy_from_slice(b"If you shed tears when you miss the sun, you also miss the stars.");
+        let config = Config::new(temp_dir.into_path());
+        let sst_factory = Arc::new(IoFactory::new(
+            config.dir_path.join(DEFAULT_SS_TABLE_PATH),
+            FileExtension::SSTable
+        )?);
+        let cache = ShardingLruCache::new(
+            config.table_cache_size,
+            16,
+            RandomState::default()
+        )?;
+        let mut vec_data = Vec::new();
+        let times = 2333;
+        let (log_loader, _, _) = LogLoader::reload(
+            config.path(),
+            (DEFAULT_WAL_PATH, Some(1)),
+            IoType::Buf,
+            |_| Ok(())
+        )?;
+        let mut log_writer = log_loader.writer(1)?;
+
+        // 填充测试数据
+        for i in 0..times {
+            let key_value = (Bytes::from(bincode::options().with_big_endian().serialize(&i)?), Some(value.clone()));
+
+            let _ = log_writer.add_record(&data_to_bytes(key_value.clone())?)?;
+            vec_data.push(key_value);
+        }
+        // 测试重复数据是否被正常覆盖
+        let repeat_data = (vec_data[0].0.clone(), None);
+        let _ = log_writer.add_record(&data_to_bytes(repeat_data.clone())?)?;
+        vec_data[0] = repeat_data.clone();
+
+        log_writer.flush()?;
+
+        let sst_loader = SSTableLoader::new(
+            config,
+            sst_factory.clone(),
+            log_loader.clone()
+        )?;
+
+        let _ = sst_loader.create(1, vec_data.clone(), 0)?;
+
+        assert!(sst_loader.remove(&1).is_some());
+        assert!(sst_loader.is_emtpy());
+
+        let ss_table_loaded = sst_loader.get(1).unwrap();
+
+        assert_eq!(ss_table_loaded.query_with_key(&repeat_data.0, &cache)?, repeat_data.1);
+        for i in 1..times {
+            assert_eq!(ss_table_loaded.query_with_key(&vec_data[i].0, &cache)?, Some(value.clone()))
+        }
+
+        // 模拟SSTable异常而使用Wal进行恢复的情况
+        assert!(sst_loader.remove(&1).is_some());
+        assert!(sst_loader.is_emtpy());
+        sst_loader.clean_only_sst(1).unwrap();
```

Comment:
`clean_only_sst func` These types of functions should be placed in the test module.

Reply:
> `clean_only_sst func` These types of functions should be placed in the test module.

done